pkg/build/builder/docker.go

```diff
@@ -87,6 +87,29 @@ func (d *DockerBuilder) Build() error {
 	}
 
 	buildTag := randomBuildTag(d.build.Namespace, d.build.Name)
+	imageName := d.getDockerfileFrom(buildDir)
+	if imageName == "" {
+		return fmt.Errorf("Builder image id empty.")
+	}
+	var imageExists bool = true
+	_, err = d.dockerClient.InspectImage(imageName)
+	if err != nil {
+		imageExists = false
+	}
+	// if forcePull or the image not exists on the node we should pull the image first
+	if d.build.Spec.Strategy.DockerStrategy.ForcePull || !imageExists {
+		pullAuthConfig, authPresent := dockercfg.NewHelper().GetDockerAuth(
+			imageName,
+			dockercfg.PullAuthType,
+		)
+		if authPresent {
+			glog.V(4).Infof("Authenticating Docker pull with user %q", pullAuthConfig.Username)
```

Comment:
If we don't display secret data elsewhere, maybe it's not a good idea to display the username here.